src/java.base/share/classes/sun/net/www/protocol/http/DigestAuthentication.java

```diff
@@ -427,6 +508,24 @@ public void checkResponse (String header, String method, URL url)
         checkResponse (header, method, url.getFile());
     }
 
+    private void validateAlgorithm(String algorithm) throws IOException {
+        if (getAuthType() == AuthCacheValue.Type.Server &&
+                getProtocolScheme().equals("https")) {
+            // HTTPS server authentication can use any algorithm
```

Comment:
Hello Michael,
Should it be noted somewhere in the text of the linked CSR that the property values play no role if the request URL's scheme is HTTPS (and proxy isn't involved)?

As a related question - should the implementation even bother about this HTTP vs HTTPS protocol check? As far as I understand the reason why we are proposing to disable support for MD5 is because of its cryptographic weakness/vulnerabilities. So irrespective of whether or not the  computed MD5 (or any other disabled algorithm) digest value gets transferred over HTTP or HTTPS, the value would still be cryptographically vulnerable isn't it?

Reply:
The distinction may be worth mentioning in the CSR and the docs. The distinction is useful because like Basic authentication which is completely insecure with plaintext HTTP, it is secure and a potentially useful web authentication scheme controlled by the browser when run over HTTPS (except when authenticating with HTTPS to a proxy).